src/python-common/ceph/deployment/service_spec.py

```diff
@@ -1256,6 +1257,7 @@ def __init__(self,
             extra_container_args=extra_container_args, custom_configs=custom_configs)
 
         self.initial_admin_password = initial_admin_password
+        self.protocol = protocol
```

Comment:
can we have some verification on this setting? It should only be allowed to be http or https I think. Right now you can pass any string here and while it might be handled later somewhere, I think we should error out at apply if the user passes something unexpected.

Reply:
Fixed